addons/stock/models/stock_quant.py

```diff
@@ -340,20 +338,58 @@ def get_import_templates(self):
         }]
 
     @api.model
-    def _get_forbidden_fields_write(self):
-        """ Returns a list of fields user can't edit when he want to edit a quant in `inventory_mode`."""
-        return ['product_id', 'location_id', 'lot_id', 'package_id', 'owner_id']
+    def _trigger_inventory_moves(vals):
+        return any(field in vals for field in ['location_id', 'lot_id', 'package_id', 'owner_id'])
+
+    @api.model
+    def _get_specific_fields_write(self):
+        """ Model method.
+        Return a list of field for which a balanced pair of move are created on edition.
+        :return: List of String
+        """
+        return ['location_id', 'lot_id', 'package_id', 'owner_id']
+
+    def _update_quant_by_moves(self, vals):
+        """ Handle update of more specific fields"""
+        # if any key of vals in specific fields => Move+ & Move- with all values
+
+        move_vals = list(chain.from_iterable([
+            # 'Undo' actual quant
+            quant._get_inventory_move_values(
+                quant.quantity,
+                quant.location_id,
+                quant.product_id.with_company(quant.company_id).property_stock_inventory,
+                package_id=quant.package_id,
+                product_id=quant.product_id,
+                lot_id=quant.lot_id,
+                owner_id=quant.owner_id,
+            ),
+            # 'Do' new quant
+            quant._get_inventory_move_values(
+                vals.get('quantity', quant.quantity),
+                quant.product_id.with_company(quant.company_id).property_stock_inventory,
+                self.env['stock.location'].browse(vals['location_id']) if 'location_id' in vals else quant.location_id,  # Fait peter un test
+                package_dest_id=self.env['stock.quant.package'].browse(vals['package_id']) if 'package_id' in vals else quant.package_id,
+                product_id=self.env['product.product'].browse(vals['product_id']) if 'product_id' in vals else quant.product_id,
```

Comment:
`product_id` is not in the `_trigger_inventory_moves()` but we take it in `vals`. Intended ?